tomography_preprocessing/denoising/care4relion/_utils.py

```diff
@@ -0,0 +1,107 @@
+from pathlib import Path
+from typing import Tuple, List, Dict
+
+import pandas as pd
+import starfile
+import typer
+import json
+import shutil
+
+def create_denoised_tomograms_dir(
+        output_directory: Path
+) -> Tuple[Path, Path, Path]:
+    training_dir = output_directory / 'external' / 'training' 
+    training_dir.mkdir(parents=True, exist_ok=True)
+    tomogram_dir = output_directory / 'tomograms'
+    tomogram_dir.mkdir(parents=True, exist_ok=True)
+    tilt_series_dir = output_directory / 'tilt_series'
+    tilt_series_dir.mkdir(parents=True, exist_ok=True)
+    return training_dir, tomogram_dir, tilt_series_dir
+    
+def parse_training_tomograms(
+        training_tomograms: str
+) -> List:
+    training_tomograms = training_tomograms.strip().split(':')
+    return training_tomograms
+
+def generate_training_tomograms_star(
+        global_star: pd.DataFrame,
+        training_tomograms: List,
+) -> pd.DataFrame:
+    training_tomograms_idx = pd.DataFrame(global_star.rlnTomoName.tolist()).isin(training_tomograms).values
+    if not any(training_tomograms_idx):
+        e = f"Could not user specified training tomograms ({', '.join(str(x) for x in training_tomograms)}) in tilt series star file"
+        console.log(f'ERROR: {e}')
+        raise RuntimeError(e)
+    training_tomograms_star = global_star[training_tomograms_idx]
+    return training_tomograms_star
+    
+def find_tomogram_halves(
+        training_tomograms_star: pd.DataFrame,
+) -> Tuple[List, List]:
+    even_tomos = [f'{rows.rlnTomoTomogramHalvesForDenoisingEven}' for idx,rows in training_tomograms_star.iterrows()]
+    odd_tomos = [f'{rows.rlnTomoTomogramHalvesForDenoisingOdd}' for idx,rows in training_tomograms_star.iterrows()]
+    return even_tomos, odd_tomos
```

Comment:
```suggestion
return training_tomograms_star['rlnTomoTomogramHalvesForDenoisingEven'], training_tomograms_star['rlnTomoTomogramHalvesForDenoisingOdd'] 
```

not work here?

Reply:
Opps, yep! It needs to output as a list so I'm adding .values.list() at the end of each dataframe too